keras_cv/models/task.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for Task models."""
+
+import os
+
+from tensorflow import keras
+
+from keras_cv.utils.python_utils import classproperty
+from keras_cv.utils.python_utils import format_docstring
+
+
+@keras.utils.register_keras_serializable(package="keras_cv")
+class Task(keras.Model):
```

Comment:
Do we really need this ? Just add Experiment and we have replicated all the complexity of the MG "framework".

Reply:
I don't think that's a good analogy. The `Task` is not user-facing and mainly a way to collect common methods across the codebase. We had a LOT of code duplication before introducing these base classes in KerasNLP. See https://github.com/keras-team/keras-nlp/issues/530